master/freebsd_ci.py

```diff
@@ -0,0 +1,220 @@
+        env={
+            'WORKERNAME': util.Property('workername'),
+            'MAKE_JOBS_NUMBER': util.Property('make_jobs'),
+        },
+    ),
+])
+
+###############################################################################
+#  Builders
+###############################################################################
+
+freebsdci_builder_names = {
+    'main': 'freebsdci',
+    'test': 'freebsdci_test',
+}
+
+freebsdci_builders = [
+    util.BuilderConfig(
+        name=freebsdci_builder_names['main'],
+        workernames=['gaebolg', 'abeing'],
+        factory=freebsdci_factory),
+
+    util.BuilderConfig(
+        name=freebsdci_builder_names['test'],
+        workernames=[],
+        factory=freebsdci_test_factory),
+]
+
+###############################################################################
+#  Schedulers
+###############################################################################
+
+freebsdci_schedulers = [
+    schedulers.SingleBranchScheduler(
+        name="freebsdci_master",
+        change_filter=util.ChangeFilter(branch='master'),
+        treeStableTimer=60,
+        builderNames=[freebsdci_builder_names['main']]),
+
+    schedulers.ForceScheduler(
+        name="freebsdci_force",
+        builderNames=[freebsdci_builder_names['main']]),
+
+    schedulers.ForceScheduler(
+        name="freebsdci_force_test",
+        builderNames=[freebsdci_builder_names['test']]),
+
+    schedulers.SingleBranchScheduler(
+        name='freebsdci_pull_request',
+        change_filter=util.ChangeFilter(
+            branch_re='^refs/pull/.*',
+            repository_re='https://github.com/[\w]+/julia'),
+        builderNames=[freebsdci_builder_names['main']],
+        treeStableTimer=10,
+    ),
+]
+
+###############################################################################
+#  GitHub Status Reporter
+###############################################################################
+
+freebsdci_report = reporters.GitHubStatusPush(
+    token=FREEBSDCI_OAUTH_TOKEN,
+    context='julia freebsd ci',
+    startDescription='Build started',
+    endDescription='Build done')
+
+###############################################################################
+#  GitHub Change Hook
+###############################################################################
+
+freebsdci_gh_setting = {
+    'skips': [
+        r'\[ *skip *ci *\]',
+        r'\[ *ci *skip *\]',
+        r'\[ *skip *bsd *\]',
+        r'\[ *bsd *skip *\]'
+    ],
+    'github_property_whitelist': ['base'],
+}
+
+###############################################################################
+#  Master Config
+###############################################################################
+
+c['workers'].extend(freebsdci_workers)
+c['services'].append(freebsdci_report)
+c['builders'].extend(freebsdci_builders)
+c['schedulers'].extend(freebsdci_schedulers)
+c['www']['change_hook_dialects']['github'].update(freebsdci_gh_setting)
```

Comment:
Will this bring some bad side effect?
Or how to creating another GitHub dialect?

Reply:
This is a good thing to add to the change hook dialect that is already in master.cfg

Reply:
There are some CI only stuffs in `freebsdci_gh_setting`, like `skips`.
:thinking: I think we should have different setting for CI.

Reply:
I'm not sure what you mean; The FreeBSD CI and the Linux/Windows/Mac CI should all behave the same, IMO.  I think our "CI" and our "packaging" should all be the same thing.

Reply:
e.g. https://github.com/JuliaCI/julia-buildbot/pull/87/files#diff-43e4fe43d9ca1e0a77951b0b5fecfc40R216
FreeBSD CI recognizes `bsd skip` pattern and skipping commit about that, but other builders won't want to skip this commit.

Reply:
So my question is why have a `bsd skip` pattern at all?  We should just have `ci skip` and have that apply to all of our CI (including the windows and mac and linux builds).

Reply:
The original idea is about just imitating AppVeyor (there is a pattern `av skip` for windows build).

Reply:
That exists mostly because appveyor builds are slow and unreliable.  Those are not attributes of the FreeBSD CI, which is fast and reliable.  :)  I think we should just limit it to `ci skip` and leave it at that; I don't see any reason to differentiate our builds by platform like that here.  If someone has a usecase for this that I'm not seeing, I'd like to see it.

Reply:
Ah, fine. Let's drop the `bsd skip` pattern.